### PR TITLE
Refund on deleting rooms & update popup

### PR DESCRIPTION
**Pull request by khatakslied**

### CODE REVIEW

**Comment by khatakslied** [comment]
The pop up still needs styling, but I think this is a step in the right direction.

File: scripts/popup/GUIPopup.gd [hunk 1]
```diff
@@ -3,18 +3,31 @@
...
+	return self
+
+func set_yes_text(text: String) -> GUIPopup:
+	yes.text = text
+	return self
+
+func set_no_text(text: String) -> GUIPopup:
+	no.text = text
+	return self
+
```
Comment (khatakslied) [inline scripts/popup/GUIPopup.gd:30]: Separated the title text from the body text (perhaps we can rename it body_text, actually). And added text setters for the yes & no buttons.

File: scripts/scene/BuildingManager.gd [hunk 2]
```diff
@@ -136,7 +136,7 @@ func _on_setting_door_state_input(event: InputEvent) -> void:
 		room_builder.setting_door_motion(event, camera.position, camera.zoom)
 
 func _on_confirming_room_state_entered() -> void:
-	popup.set_text(room_builder.popup_message).show()
+	popup.set_title(room_builder.popup_title).set_content(room_builder.popup_content).set_yes_text(room_builder.popup_yes_text).set_no_text(room_builder.popup_no_text).show()
```
Comment (khatakslied) [inline scripts/scene/BuildingManager.gd:139]: No need to pass the text on instantiating a popup (new_popup()). We can access these functions on through the room_builder/room_editor instead.

File: scripts/scene/RoomBuilder.gd [hunk 3]
```diff
@@ -181,6 +190,7 @@ func draw_rooms() -> void:
 	# Clear drafting layer
 	build_tile_map.clear_layer(drafting_layer)
 	build_tile_map.clear_layer(building_layer)
+	furniture_tile_map.clear_layer(hotspot_layer)
```
Comment (khatakslied) [inline scripts/scene/RoomBuilder.gd:193]: I missed this in the last commit, but the hotspots were not being cleared on deleting a room.

**Review by conalli — APPROVED** [review]
Nice! Looks good

File: scripts/scene/RoomBuilder.gd (hunk 3 above)
Comment (conalli) [reply scripts/scene/RoomBuilder.gd:193]: 🙌 was about to make an issue but you caught it already

**Pull request merged**